state/postgresql/v2/postgresql.go

```diff
@@ -68,28 +76,75 @@ func NewPostgreSQLStateStoreWithOptions(logger logger.Logger, opts Options) stat
 	s := &PostgreSQL{
 		logger:        logger,
 		enableAzureAD: !opts.NoAzureAD,
+		enableAWSIAM:  !opts.NoAWSIAM,
 	}
 	s.BulkStore = state.NewDefaultBulkStore(s)
 	return s
 }
 
 // Init sets up Postgres connection and performs migrations
 func (p *PostgreSQL) Init(ctx context.Context, meta state.Metadata) error {
-	err := p.metadata.InitWithMetadata(meta, p.enableAzureAD)
-	if err != nil {
-		p.logger.Errorf("Failed to parse metadata: %v", err)
-		return err
+	var (
+		useAWS   bool
+		useAzure bool
+		err      error
+	)
+	if meta.Properties["UseAWSIAM"] == "" {
```

Comment:
nit: this would look cleaner as a switch

Reply:
Its really a nit, but decreases the {}
```
switch meta.Properties["UseAWSIAM"] {
case "":
...
default:
...
}
```

Reply:
We should make sure that all properties are read as case-insensitive. So we can't read from meta.Properties directly. Use the metadata.GetMetadataProperty method which is case-insensitive.